src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -12,109 +12,49 @@
-
-            public bool Equals(DefinitionProject other)
-                => EqualityComparer<ProjectId?>.Default.Equals(_sourceProjectId, other._sourceProjectId) &&
-                   _assemblyName == other._assemblyName;
-
-            public override int GetHashCode()
-                => Hash.Combine(_sourceProjectId, _assemblyName?.GetHashCode() ?? 0);
-        }
-
-        private readonly struct DependentProject : IEquatable<DependentProject>
-        {
-            public readonly ProjectId ProjectId;
-            public readonly bool HasInternalsAccess;
-
-            public DependentProject(ProjectId dependentProjectId, bool hasInternalsAccess)
-            {
-                this.ProjectId = dependentProjectId;
-                this.HasInternalsAccess = hasInternalsAccess;
-            }
-
-            public override bool Equals(object? obj)
-                => obj is DependentProject project && this.Equals(project);
-
-            public override int GetHashCode()
-                => Hash.Combine(HasInternalsAccess, ProjectId.GetHashCode());
-
-            public bool Equals(DependentProject other)
-                => HasInternalsAccess == other.HasInternalsAccess && ProjectId.Equals(other.ProjectId);
-        }
-
         /// <summary>
         /// Dependent projects cache.
-        /// For a given solution, maps from an assembly (source/metadata) to the set of projects referencing it.
-        ///     Key: DefinitionProject, which contains the assembly name and a flag indicating whether assembly is source or metadata assembly.
+        /// For a given solution, maps from an assembly (source-project or metadata-assembly) to the set of projects referencing it.
+        ///     Key: DefinitionProject, which contains the project-id for a source-project-assembly, or assembly-name for a metadata-assembly.
         ///     Value: List of DependentProjects, where each DependentProject contains a dependent project ID and a flag indicating whether the dependent project has internals access to definition project.
         /// </summary>
-        private static readonly ConditionalWeakTable<Solution, ConcurrentDictionary<DefinitionProject, ImmutableArray<DependentProject>>> s_dependentProjectsCache =
-            new ConditionalWeakTable<Solution, ConcurrentDictionary<DefinitionProject, ImmutableArray<DependentProject>>>();
-
-        /// <summary>
-        /// Used to create a new concurrent dependent projects map for a given assembly when needed.
-        /// </summary>
-        private static readonly ConditionalWeakTable<Solution, ConcurrentDictionary<DefinitionProject, ImmutableArray<DependentProject>>>.CreateValueCallback s_createDependentProjectsMapCallback =
-            _ => new ConcurrentDictionary<DefinitionProject, ImmutableArray<DependentProject>>(concurrencyLevel: 2, capacity: 20);
+        private static readonly ConditionalWeakTable<Solution, DependentProjectMap> s_dependentProjectsCache =
```

Comment:
I'd be really curious what the hit rate is of this cache in the end...

Reply:
100% when you perform multiple operations in a row.  and it's a massive speedup.  i will have numbers tomorrow.

Reply:
Actually, i'm thinking of other caches.  I think this cache is not necessary.  But i'm going to remove in a future change (i'm making several changes here).

Reply:
That was my thinking: given it's keyed on Solution it's only going to matter for multiple searches when no changes happened in the middle, and if the key is a project/assembly pair it has to be multiple searches _originating from the same place_. Was just curious, but if we think it's worth it I'm not against keeping it.

Reply:
i no longer think ti's important based on perf numbers i'm seeing.  But i'm going to hold off on removing to an independent PR where i can validate making just that change :)